daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -20,8 +21,7 @@ final class Hash private (private val bytes: Array[Byte]) {
 
   def toByteArray: Array[Byte] = bytes.clone()
 
-  def toHexaString: String =
-    bytes.map("%02x" format _).mkString
+  def toHexaString: String = BaseEncoding.base16().encode(bytes)
```

Comment:
You can probably pull out `BaseEncoding.base16()` into a `private val hexEncoding` for **even more** performance.